src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeMessagingController.java

```diff
@@ -129,19 +133,35 @@ public void saveListResponse(
         }
     }
 
-    @RequestMapping(value = "/api/v1/data-nodes/atlas", method = POST)
-    public void addAtlasInformation(@RequestBody @Valid AtlasInfoDTO atlasInfoDTO, Principal principal)
-            throws PermissionDeniedException, NotExistException {
+    @RequestMapping(value = "/api/v1/data-nodes/atlases", method = POST)
+    public AtlasShortDTO createOrUpdateAtlas(
+            @RequestBody @Valid AtlasShortDTO atlasShortDTO,
+            Principal principal
+    ) throws PermissionDeniedException, NotExistException {
 
-        final DN datanode = getDatanode(principal);
-        String atlasVersion = null;
-        if (atlasInfoDTO.getInstalled()) {
-            atlasVersion = atlasInfoDTO.getVersion();
-        }
-        if (!Objects.equals(datanode.getAtlasVersion(), atlasVersion)) {
-            datanode.setAtlasVersion(atlasVersion);
-            baseDataNodeService.updateAtlasInfo(datanode);
+        A atlas = convert(atlasShortDTO);
+        A updated;
+
+        if (atlas.getId() == null || atlasService.findByIdInAnyTenant(atlas.getId()) == null) {
+            final DN dataNode = getDatanode(principal);
+
```

Comment:
Just to clarify, if Atlas exists but not assigned to any tenant one would be saved twice. Is that expected behavior?

Reply:
I've added Javadoc for clariication of method naming